src/linux_mcp_server/commands.py

```diff
@@ -0,0 +1,196 @@
+    ),
+    # File content
+    "read_file": CommandSpec(args=["cat", "{path}"]),
+    # === Multi-command tools (CommandGroup) ===
+    "network_interfaces": CommandGroup(
+        commands={
+            "brief": CommandSpec(args=["ip", "-brief", "address"]),
+            "detail": CommandSpec(args=["ip", "address"]),
+            "stats": CommandSpec(args=["cat", "/proc/net/dev"]),
+        }
+    ),
+    "system_info": CommandGroup(
+        commands={
+            "hostname": CommandSpec(args=["hostname"]),
+            "os_release": CommandSpec(args=["cat", "/etc/os-release"]),
+            "kernel": CommandSpec(args=["uname", "-r"]),
+            "arch": CommandSpec(args=["uname", "-m"]),
+            "uptime": CommandSpec(args=["uptime", "-p"]),
+            "boot_time": CommandSpec(args=["uptime", "-s"]),
+        }
+    ),
+    "cpu_info": CommandGroup(
+        commands={
+            "model": CommandSpec(args=["grep", "-m", "1", "model name", "/proc/cpuinfo"]),
+            "logical_cores": CommandSpec(args=["grep", "-c", "^processor", "/proc/cpuinfo"]),
+            "physical_cores": CommandSpec(args=["grep", "^core id", "/proc/cpuinfo"]),
+            "frequency": CommandSpec(args=["grep", "-m", "1", "cpu MHz", "/proc/cpuinfo"]),
+            "load_avg": CommandSpec(args=["cat", "/proc/loadavg"]),
+            "top_snapshot": CommandSpec(args=["top", "-bn1"]),
+        }
+    ),
+    "memory_info": CommandGroup(
+        commands={
+            "free": CommandSpec(args=["free", "-b"]),
+        }
+    ),
+    "process_info": CommandGroup(
+        commands={
+            "ps_detail": CommandSpec(
+                args=["ps", "-p", "{pid}", "-o", "pid,user,stat,pcpu,pmem,vsz,rss,etime,comm,args"]
+            ),
+            "proc_status": CommandSpec(args=["cat", "/proc/{pid}/status"]),
+        }
+    ),
+    "hardware_info": CommandGroup(
+        commands={
+            "lscpu": CommandSpec(args=["lscpu"]),
+            "lspci": CommandSpec(args=["lspci"]),
+            "lsusb": CommandSpec(args=["lsusb"]),
+        }
+    ),
+}
+
+
+def get_command(name: str) -> CommandEntry:
+    """Get a command specification by name.
+
+    Args:
+        name: The command name in the registry.
+
+    Returns:
+        The CommandSpec or CommandGroup for the given name.
+
+    Raises:
+        KeyError: If the command name is not found in the registry.
+    """
+    return COMMANDS[name]
+
+
+def substitute_command_args(args: list[str], **kwargs) -> list[str]:
+    """Substitute placeholder values in command arguments.
+
+    Args:
+        args: List of command arguments, possibly with {placeholder} values.
+        **kwargs: Key-value pairs to substitute into placeholders.
+
+    Returns:
+        List of command arguments with placeholders replaced.
+
+    Example:
+        >>> substitute_command_args(["ps", "-p", "{pid}"], pid=1234)
+        ["ps", "-p", "1234"]
+    """
+    return [arg.format(**kwargs) for arg in args]
```

Comment:
Even if this is done in lots of places, I think it would be more obvious to just do the comprehension directly instead of calling a function. I could be wrong, but this doesn't seem worth it to make this call.